root/programs/MultiPurposeAuthSite/MultiPurposeAuthSite/Models/ASPNETIdentity/Filter/OpenIDConnectModule.cs

```diff
@@ -0,0 +1,325 @@
+                {
+                    string responseType = query.Substring(query.IndexOf(pattern) + pattern.Length);
+
+                    bool a = false;
+                    bool b = false;
+
+                    a = responseType.StartsWith("id_token%20token");
+
+                    if (!a)
+                    {
+                        b = responseType.StartsWith("id_token");
+                    }
+                    
+                    if (a || b)
+                    {
+                        this.Rewrited = true;
+                        string temp = path.Substring(path.IndexOf(context.Request.ApplicationPath));
+
+                        string rewritedVirtualPath = "";
+                        this.OriginalVirtualPath = temp + query;
+
+                        if (a)
+                        {
+                            rewritedVirtualPath = temp + query.Replace("response_type=id_token%20token", "response_type=token");
+
+                        }
+                        else if (b)
+                        {
+                            rewritedVirtualPath = temp + query.Replace("response_type=id_token", "response_type=token");
+                        }
+                       
+                        context.RewritePath(rewritedVirtualPath, false);
+                        //context.RewritePath(context.Request.ApplicationPath + "?hoge=aaa");
+
+                        //context.Request.ContentEncoding = new OpenIDConnectEncoding(Encoding.GetEncoding("Shift_JIS"));
+                    }
+                }
+            }
+        }
+
+        /// <summary>OnEndRequest</summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void OnEndRequest(object sender, EventArgs e)
+        {
+            // EndRequestのロジックはここに挿入
+        }
+
+        private void OnPreRequestHandlerExecute(object sender, EventArgs e)
+        {
+            // OnPreRequestHandlerExecuteのロジックはここに挿入
+
+            if (this.Rewrited)
+            {
+                HttpApplication application = (HttpApplication)sender;
+                HttpContext context = application.Context;
+                context.RewritePath(this.OriginalVirtualPath);
+            }
+        }
+
+        /// <summary>OnPreSendRequestHeaders</summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void OnPreSendRequestHeaders(object sender, EventArgs e)
+        {
+            // PreSendRequestHeadersのロジックはここに挿入
+
+            HttpApplication application = (HttpApplication)sender;
+            HttpContext context = application.Context;
+
+            if (context.Request.Url.AbsolutePath.IndexOf(
+                ASPNETIdentityConfig.OAuthAuthorizeEndpoint) != -1)
+            {
+                bool a = (context.Request.QueryString["response_type"].ToLower() == "id_token token");
+                bool b = (context.Request.QueryString["response_type"].ToLower() == "id_token");
```

Comment:
@daisukenishino 

The name of variable should be easy to understand.
The developers might not be able to understand the uses this variable from the name called *b*.